runtime/lua/vim/_defaults.lua

```diff
@@ -486,3 +486,11 @@ if tty then
     end
   end
 end
+
+--- Default 'grepprg' to ripgrep if available.
+if vim.fn.executable('rg') == 1 then
+  -- Match :grep default, otherwise rg searches cwd by default
+  -- Use -uuu to make ripgrep not do its default filtering
+  vim.o.grepprg = 'rg --vimgrep -uuu $* ' .. (vim.fn.has('unix') == 1 and '/dev/null' or 'nul')
```

Comment:
Apologies for the late review, but what is the rationale for `-uuu` rather than `-uu`? Is this strictly for compatibility with `grep` (which also searches binary files by default)?

Omitting binary file matches strikes me as a better default for a text editor. If the concern is consistent behavior for both `rg` and `grep`, I'd suggest adding `-I` to the default `grepprg` to ignore binary file matches in all cases.

Reply:
yes rg -uuu matches grep -r. I do agree. The documentation at least points out that the user may wish to adjust the flags used

Reply:
The use of `/dev/null` here is also unnecessary AFAIK. It's used for grep because some versions of grep (e.g. BSD grep) will try to read from stdin if no file argument is given, so a plain `grep foo` will block indefinitely and the `/dev/null` argument prevents that.

But ripgrep does not behave that way, and adding `/dev/null` here means you cannot use `:grep foo` to search for foo recursively (since that expands to `rg -uuu foo /dev/null`, which of course returns nothing).

Reply:
ripgrep searches the current directory by default. given it's also recursive by default, the /dev/null placeholder is to be both consistent with regular grep, and to prevent an accidental grep without files from potentially spinning up a lengthy process.